src/main/java/io/fusionauth/http/server/HTTPRequestProcessor.java

```diff
@@ -34,8 +36,8 @@
 public class HTTPRequestProcessor {
   private final int bufferSize;
 
-  // TODO : Should this be sized with a configuration parameter?
-  private final StringBuilder builder = new StringBuilder();
+  // Allocate a 4k buffer for starters, it will grow as needed.
+  private final ByteArrayOutputStream byteBuffer = new ByteArrayOutputStream(4096);
```

Comment:
I suppose this should be ok? We will be allocating at least 4k on each request - this is quite a bit more than the previous `StringBuilder` - but I suppose that was likely having to re-allocate the underlying array right away. 

Perhaps it is very common for the HTTP request preamble to exceed 4k? Any performance concern on this change?

Reply:
I presume the previous approach did a lot of re-allocations and copying while it was parsing the prelude. This _feels_ large enough to not need re-allocation in most cases, while not going overboard.